10_datastructures_lab/01.py

```diff
@@ -1,19 +1,13 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-
-When valid credentials are entered print:
-    Welcome Master
-
-And when invalid credentials are entered print:
-    INTRUDER ALERT
-
-and exit the program with a non-zero exit code
-"""
-
+#!/usr/bin/python
+import sys , os
+username_password = {'apple' : 'red' ,'lettuce': 'green' ,'lemon':'yellow','orange': 'orange'}
+username = raw_input('Input username: ')
+if username not in username_password:
+	print 'INTRUDER ALERT'
+	sys.exit() 
+else:
+	password = raw_input('Input password: ')
+	if username_password[username] == password:
+		print 'Welcome Master'
+	else:
+		print 'INTRUDER ALERT'
```

Comment:
Good work. Some thoughts:

1. I think it's better to ask for username AND password in advance, and then check the data. That way intruders will have harder time guessing user/password pairs

2. The message your write in sys.exit will appear when program exits, so you can use `sys.exit('INTRUDER ALERT')`

3. Though I didn't talk about exceptions yet (they'll appear in a future chapter), it may be nice to see a solution with try/except, so I'll just put it here:

```
import sys
saved_password = {'apple' : 'red' ,'lettuce': 'green' ,'lemon':'yellow','orange': 'orange'}

username = raw_input('Enter Username: ')
password = raw_input('Enter Password: ')

try:
  if saved_password[username] != password:
    raise Exception()
except Exception:
    sys.exit('INTRUDER ALERT')

print 'Welcome, Master'
```